cdap-app-fabric/src/main/java/co/cask/cdap/gateway/handlers/WorkflowHttpHandler.java

```diff
@@ -114,9 +115,9 @@
                       QueueAdmin queueAdmin, Scheduler scheduler, PreferencesStore preferencesStore,
                       MRJobInfoFetcher mrJobInfoFetcher, ProgramLifecycleService lifecycleService,
                       MetricStore metricStore, DatasetFramework datasetFramework,
-                      DiscoveryServiceClient discoveryServiceClient) {
+                      DiscoveryServiceClient discoveryServiceClient, RouteStore routeStore) {
     super(store, runtimeService, discoveryServiceClient, lifecycleService, queueAdmin, scheduler, preferencesStore,
-          mrJobInfoFetcher, metricStore);
+          mrJobInfoFetcher, metricStore, routeStore);
```

Comment:
This is one of the reason why the new handler methods shouldn't be in the `ProgramLifecycleHandler`.